sched/scheduler/stateful_scheduler.go

```diff
@@ -412,3 +434,53 @@ func (s *statefulScheduler) scheduleTasks() {
 			})
 	}
 }
+
+/**
+If the job is pending or in progress kill it.
+*/
+func (s *statefulScheduler) KillJob(jobId string) error {
```

Comment:
This allows inProgressJobs to be read and written from different threads, which isn't safe. This has to be done like checkJobs in my changeset, i.e. with a resultCh.